### PR TITLE
add sitemap creation documentation

### PR DESCRIPTION
**Pull request by LukeSchlangen**
Adding a short explanation for why you might want a sitemap and how to add a sitemap to the project.

### CODE REVIEW

**Comment by LukeSchlangen** [comment]
I'm not sure why this failed? Would appreciate any insight.

**Comment by pieh** [comment]
> I'm not sure why this failed? Would appreciate any insight.

This fail is not related to the PR, please don't worry about it

File: docs/docs/creating-a-sitemap.md [hunk 1]
```diff
@@ -2,7 +2,23 @@
...
+
+## How to Use
+
+```javascript
+// In your gatsby-config.js
+siteMetadata: {
+  siteUrl: `https://www.example.com`,
+},
+plugins: [`gatsby-plugin-sitemap`]
+```
```
[the remark this answers is not in the document]
Comment (LukeSchlangen) [reply docs/docs/creating-a-sitemap.md]: Cool! I didn't know you could do this! Created a new commit to use this.

File: docs/docs/creating-a-sitemap.md [hunk 2]
```diff
@@ -2,7 +2,26 @@
...
+	siteUrl: `https://www.example.com`,
+  },
+  plugins: [
+    `gatsby-plugin-sitemap`
+  ]
+}
+```
+
+Above is the minimal configuration required to have it work. By default, the
+generated sitemap will include all of your site's pages, except the ones you exclude.
```
Comment (pieh) [inline docs/docs/creating-a-sitemap.md]:
```suggestion
generated sitemap will include all of your site's pages.
```
By default we don't really exclude anything :)

File: docs/docs/creating-a-sitemap.md [hunk 3]
```diff
@@ -2,7 +2,26 @@
...
+## Install gatsby-plugin-sitemap
+
+`npm install --save gatsby-plugin-sitemap`
+
+## How to Use
+
+```javascript{3,6}:title=gatsby-config.js
+module.exports = {
+  siteMetadata: {
+	siteUrl: `https://www.example.com`,
```
Comment (pieh) [inline docs/docs/creating-a-sitemap.md]: formatting is a bit off here - it should be indented more

Comment (LukeSchlangen) [reply docs/docs/creating-a-sitemap.md]: I can definitely change that. Should it be 4 spaces per indentation?

Comment (shannonbux) [reply docs/docs/creating-a-sitemap.md]: Have you tried using the Prettier extension with VS code? That's the only way I can figure out formatting :)

**Comment by LekoArts** [comment]
Sorry, seems like this PR got overlooked and another one got merged in the meantime: https://github.com/gatsbyjs/gatsby/pull/10106

We'd be happy to see improvements from you, so have a look if you want (e.g. the "I think it would be good idea to mention that siteMatadata.siteUrl is used as base for urls generated in sitemap?" part).

Thanks for the PR 👍

**Pull request closed without merging**